flask_resty/pagination.py

```diff
@@ -271,49 +294,27 @@ def ensure_query_sorting(self, query, view):
         sorting criterion, then returns the field orderings for use in the
         other methods, as in `get_field_orderings` below.
 
-        :param query: The query to paginate.
-        :type query: :py:class:`sqlalchemy.orm.query.Query`
         :param view: The view with the model we wish to paginate.
         :type view: :py:class:`ModelView`
-        :return: The sorted query & the field orderings
-        :rtype: tuple
+        :return: The field orderings necessary to do cursor pagination deterministically
+        :rtype: FieldOrderings
         """
-        (
-            sorting_field_orderings,
-            missing_field_orderings,
-        ) = self.get_sorting_and_missing_field_orderings(view)
 
-        query = view.sorting.sort_query_by_fields(
-            query, view, missing_field_orderings
-        )
-        field_orderings = sorting_field_orderings + missing_field_orderings
-
-        return query, field_orderings
+        # ignore the passed in sort so that it's consistent
+        # with further calls in get_page
+        return self.get_field_orderings(view)
```

Comment:
yeah this is bizarre too, as pagination and sorting contend for who sorts... I guess it's ok. can't say it's clean. but it's a slight change of default behavior, even for non-backward requests. not sure if actually breaks anything

Reply:
do we think it's worth making a backwards compatible version here? This seems like it should be fine? I think that if previously the sorts were different between pagination and sorting it would have been wrong right?